cmp/grid/GridModel.js

```diff
@@ -350,66 +344,67 @@ export class GridModel {
     }
 
     noteAgColumnStateChanged(agColState) {
-        const colChanges = agColState.map(({colId, width, hide}) => {
+        const colStateChanges = agColState.map(({colId, width, hide}) => {
             const col = this.findColumn(this.columns, colId);
+            if (!col) return null;
             return {
                 colId,
                 hidden: hide,
                 width: col.flex ? undefined : width
             };
         });
 
-        this.applyColumnChanges(colChanges);
+        pull(colStateChanges, null);
+        this.applyColumnStateChanges(colStateChanges);
     }
 
     /**
      * This method will update the current column definition. Throws an exception if any of the
-     * columns provided in colChanges are not present in the current column list.
+     * columns provided in colStateChanges are not present in the current column list.
      *
      * Note: Column ordering is determined by the individual (leaf-level) columns in state.
      * This means that if a column has been redefined to a new column group, that entire group may
      * be moved to a new index.
      *
-     * @param {ColumnState[]} colChanges - changes to apply to the columns. If all leaf columns are
+     * @param {ColumnState[]} colStateChanges - changes to apply to the columns. If all leaf columns are
      *      represented in these changes then the sort order will be applied as well.
      */
     @action
-    applyColumnChanges(colChanges) {
-        let {columns} = this,
-            newCols = [...columns];
+    applyColumnStateChanges(colStateChanges) {
+        let columnState = [...this.columnState];
```

Comment:
Yes we are only cloning here in order to trigger the reaction.

Agree that a deep clone is probably more appropriate, especially now that we are dealing with the simple column state objects instead of Columns